network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/NetworkObjectIndex.java

```diff
@@ -1067,6 +1067,41 @@ public Identifiable<?> getIdentifiable(String id) {
         return null;
     }
 
+    public AbstractInjectionImpl getRegulatingEquipment(String id, ResourceType resourceType) {
```

Comment:
because the list of regulatingEquipments contains equipments that regulateit is only svc, generator, vsc convertater stations and shunt compensator (and twt but I didnt add them in these pr)